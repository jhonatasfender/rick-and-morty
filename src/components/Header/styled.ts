import { Link } from 'react-router-dom'
import styled, { css } from 'styled-components'

type WrapperProps = {
  fixed: boolean
}

export const Wrapper = styled.nav<WrapperProps>`
  ${({ theme, fixed }) => css`
    display: flex;
    justify-content: center;
    position: ${fixed ? 'fixed' : 'inherit'};
    width: 100%;
    height: 9rem;
    padding: 3rem;
    background-color: ${theme.colors.backgroundHeader};
    z-index: ${theme.layers.menu};
  `}
`

export const Menu = styled.ul`
  ${({ theme }) => css`
    display: flex;
    color: ${theme.colors.white};
    font-size: 2.9rem;
    font-weight: bold;
    list-style-type: none;
    gap: 8rem;
  `}
`

type ItemMeuProps = {
  active?: boolean
}

export const ItemMeu = styled.li<ItemMeuProps>`
  ${({ active, theme }) => css`
    user-select: none;
    cursor: pointer;

    ${active &&
    css`
      &::after {
        content: '';
        background-color: ${theme.colors.redOrange};
        display: block;
        width: calc(100% + 1rem);
        margin-left: -0.5rem;
        height: 1rem;
        border-radius: 0.4rem;
      }
    `}
  `}
`

export const LinkWrapper = styled(Link)`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    text-decoration: none;
  `}
`
