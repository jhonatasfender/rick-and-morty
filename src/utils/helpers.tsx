import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { render, RenderResult } from '@testing-library/react'

import theme from '../styles/theme'

const queryClient = new QueryClient()

export const renderWithThemeAndRoute = (children: ReactNode): RenderResult =>
  render(
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </MemoryRouter>
    </QueryClientProvider>,
  )
