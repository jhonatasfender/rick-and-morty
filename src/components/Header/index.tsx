import { useLocation } from 'react-router-dom'
import * as S from './styled'

export default function Header(): JSX.Element {
  const { pathname } = useLocation()

  return (
    <S.Wrapper fixed={pathname === '/'}>
      <S.Menu>
        <S.ItemMeu active={pathname === '/'}>
          <S.LinkWrapper to="/">Home</S.LinkWrapper>
        </S.ItemMeu>
        <S.ItemMeu active={pathname.includes('/character')}>
          <S.LinkWrapper to="/character">Character</S.LinkWrapper>
        </S.ItemMeu>
      </S.Menu>
    </S.Wrapper>
  )
}
