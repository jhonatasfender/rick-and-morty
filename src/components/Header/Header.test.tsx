import { renderWithThemeAndRoute } from '../../utils/helpers'
import Header from './'

describe('<Header />', () => {
  it('should load the component with default values', () => {
    const { container } = renderWithThemeAndRoute(<Header />)

    expect(container).toMatchSnapshot()
  })
})
