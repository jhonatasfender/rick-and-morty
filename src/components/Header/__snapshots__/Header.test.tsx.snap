// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should load the component with default values 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  width: 100%;
  height: 9rem;
  padding: 3rem;
  background-color: rgb(40 37 63 / 75%);
  z-index: 20;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #ffffff;
  font-size: 2.9rem;
  font-weight: bold;
  list-style-type: none;
  gap: 8rem;
}

.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c2::after {
  content: '';
  background-color: #F6313C;
  display: block;
  width: calc(100% + 1rem);
  margin-left: -0.5rem;
  height: 1rem;
  border-radius: 0.4rem;
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c3 {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <nav
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="c4"
      >
        <a
          class="c3"
          href="/character"
        >
          Character
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
